sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala

```diff
@@ -195,12 +195,23 @@ trait Interpreter extends ScorexLogging {
   def fullReduction(ergoTree: ErgoTree,
                     context: CTX,
                     env: ScriptEnv): (SigmaBoolean, Long) = {
+    if (context.activatedScriptVersion > Interpreter.MaxSupportedScriptVersion) {
+      // majority has already switched to higher version, accept without verification
+      // NOTE: this path should never be taken for validation of candidate blocks
+      // this means Ergo node should always pass Interpreter.MaxSupportedScriptVersion
+      // as the value of ErgoLikeContext.activatedScriptVersion.
+      // see also ErgoLikeContext ScalaDoc.
+      return TrivialProp.TrueProp -> context.initCost
+    } else {
+      if (ergoTree.version > context.activatedScriptVersion)
+        throw new InterpreterException(s"Not supported script version ${ergoTree.version}")
+    }
```

Comment:
Please also comment how this versioning logic is interacting with previously done one (trySoftForkable here and in reduceToCrypto)

Reply:
added description

Reply:
@aslesarenko there is an issue here it seems.
"context.activatedScriptVersion > Interpreter.MaxSupportedScriptVersion" will work not for the HF coming but also for future soft-forks. The trySoftForkable will not reached, and soft-forks have hardest processing possible.

Possible solutions:

1) make "context.activatedScriptVersion > Interpreter.MaxSupportedScriptVersion" condition working for the upcoming version only
2) Leave as it is, and leave TODO to change validation in 5.0 (this change will be compatible with old clients).

Reply:
Yes, I implied the second option, will leave `TODO v5.0` in the code. 
Also note: 
1)  in v5.0 MaxSupportedScriptVersion will be increased to 1.
2) in v5.0 the code will be versioned and all semantic changes in the language (due to fixes and new features) will be versioned, this is independent from this specific conditions (which will have to be reviewed again in v5.0)
3) I'm not sure about the usefulness of trySoftForkable in the future, it may be too hard to use, but will see how it goes.